textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -256,24 +273,50 @@ private ITuple joinTuples(ITuple outerTuple, ITuple innerTuple, JoinPredicate jo
 			return null;
 		}
 
-		// TODO Discuss and implement as to what Schema has to be.
-		// It shouldn't just be inner tuple or outer tuple schema.
-		Schema schema = innerTuple.getSchema();
-		List<IField> fieldList = innerTuple.getFields();
-		IField[] nextTupleField = new IField[fieldList.size()];
+		List<IField> newFieldList = new ArrayList<>();
+
+		ArrayList<Attribute> innerAttrList = new ArrayList<>();
+		innerAttrList.addAll(innerTuple.getSchema().getAttributes());
```

Comment:
Why not just
`List<Attribute> innerAttrList = innerTuple.getSchema().getAttributes();`

Moreover, operations involving schema should be done in`open()`. If we do it in `getNextTuple`, then there's a small overhead every time `getNextTuple()` is called.

@sripadks The IOperator interface is changed recently. We added a new method called `getOutputSchema()`. Each operator needs to specify its output schema. 

Please take a look at other operators, for example, KeywordMatcher, to find out how we now handles output schema.